source/controllers/ListingController.ts

```diff
@@ -115,13 +134,22 @@ listing.interface('/item', async (req, res) => {
 			}
 			const author = await User.findByPk(listing.userId);
 			if (doesUserExist(author)) {
-				res.render('pages/main/listing_item.ejs', {
-					listing,
-					author: author.username,
-					author_profile: author.username,
-					authorid: author.id,
-					constants: headerConstants
-				});
+				if ((author.banned && (isLoggedOn(res.locals.user) && !res.locals.user.accesslevel.category_admin && author.id !== res.locals.user.id)) || (author.banned && (res.locals.user === null || res.locals.user === undefined))) {
+					res.redirect('/section');
+				} else {
+					if ((listing.status === 'open') || (listing.status === 'closed' && isLoggedOn(res.locals.user) && (res.locals.user.id === author.id || res.locals.user.accesslevel.category_admin))) {
```

Comment:
very long if statements
you should split them or at least make them span multiple lines